Fixing bug that throws java.lang.NoSuchMethodError: android.test.Activity

pose opened the pull request:
Fixing bug that throws java.lang.NoSuchMethodError: android.test.ActivityInstrumentationTestCase2.<init> at least in 2.2.1 version or above.

pose committed:
Fixing bug that throws java.lang.NoSuchMethodError: android.test.ActivityInstrumentationTestCase2.<init> at least in 2.2.1 version or above.

pose commented:
Hi,

I am submitting a patch for the following bug: http://groups.google.com/group/maven-android-developers/browse_thread/thread/29930d25a3d3f66f

Thanks in advance,
Alberto

androidweasel commented:
The patch comment is misleading: the fix makes the code compatible with Android 1.5: ActivityInstrumentationTestCase2(Class) didn't exist then. (Most of the fix suggested in the Google Groups discussion cited - adding junit4, etc - is unnecessary.)

Since ActivityInstrumentationTestCase2(String, Class) is deprecated, perhaps it would be worth adding a comment to that effect, or making the example only work with 1.6 or 2.x and above.

pose commented:
Yes. I have double checked it and you are right.

cescoffier commented:
Hello,

So, I still didn't have the time to look into the patch. What's the status ?

androidweasel commented:
Although the patch gets things working, it does so by making the code run as 1.5 (or perhaps 1.1) code, using a deprecated constructor.

It may be more useful as an example to set the target SDK to 1.6 instead, since that'll highlight where the minimum or target SDK settings should be set, and nobody these days should be using the default setting of targeting _all_ SDK levels including 1.1.

pose commented:
Hello!

I have added conditional code that changes with the `platform` number required property:
- If the `platform` number is less than or equal to 7, it uses the deprecated constructor (as no other option is present and it is really misleading that the code generated by the archetype does not compile by default). 
- Otherwise, if the `platform` property is set to greater than 7, it uses the recommended and not deprecated constructor.

Moreover, not only do the unit tests are compiling and running correctly, but also I have added more assertions to detect that this conditional change has been made.

Thanks in advance,

pose commented:
Hello, 

Sorry to insist, I would really like to know if you are interested in applying this changeset.

Thanks in advance,

cescoffier commented:
You're really lucky... I'm on it !
I'm working on a second change request right now.

cescoffier commented:
Could you please check that everything works as expected ?
